framework/include/utils/RankTwoTensor.h

```diff
@@ -169,9 +203,22 @@ class RankTwoTensorTempl : public TensorValue<T>
    */
   void fillFromScalarVariable(const VariableValue & scalar_variable);
 
-public:
   /// returns _coords[i][c], ie, column c, with c = 0, 1, 2
-  TypeVector<T> column(const unsigned int c) const;
+  VectorValue<T> column(const unsigned int c) const;
+
+  /// return the matrix multiplied with its transpose A*A^T (guaranteed symmetric)
+  static RankTwoTensorTempl<T> timesTranspose(const RankTwoTensorTempl<T> &);
+
+  /// return the matrix multiplied with its transpose A^T*A (guaranteed symmetric)
+  static RankTwoTensorTempl<T> transposeTimes(const RankTwoTensorTempl<T> &);
+
+  /// return the matrix plus its transpose A+A^T (guaranteed symmetric)
+  static RankTwoTensorTempl<T> plusTranspose(const RankTwoTensorTempl<T> &);
```

Comment:
I am afraid that some users might do:
```cpp
a = b.timesTranspose(c);
```
In this case, the users are expecting `a = b * c^T`, while instead they got `a = c * c^T`.

:-/

The [[nodiscard]] attribute won't help here as nothing is discarded... I am not aware of any way to forbid calling named constructors on instances.

Reply:
Yeah, nothing can be done here, other than maybe not making this a class member at all. Here's what I found on stackexchange about this case:

> This is a thing every linter under the sun will warn about. If you try to make your API idiot-proof, the universe will conspire against you and invent a better idiot. You can't win that fight. Worry about API designs that lead your users astray or are needlessly convoluted to work with. Don't worry about attempting to stop usages that any coder 3 weeks out of uni can identify at a casual glance as clearly not how the API is intended or straight up crud code.

:-D

So, anyways, we could add a `RankTwoTools` namespace and put such methods in there.

Reply:
What a convenient quote. When you touch the code again, maybe add a sentence or two to the doxygen string warning the users what not to do.

I agree on the RankTwoTools namespace thing. Some named constructors should really be free functions instead of encapsulated methods. Future work!